### PR TITLE
Db schema

### PR DESCRIPTION
**Pull request by RamyAlshurafa**
Relates #154 
For now, all the collections created/modified except the `notification`, it must be modified later maybe in next sprint

### CODE REVIEW

File: client/src/components/Pages/InternCreateProfile/index.js [hunk 1]
```diff
@@ -259,18 +250,12 @@ export default class InternCreateProfile extends Component {
...
         const formData = {
           profileImage: { fileName: profileImage.fileName, isPrivate: false },
-          pressPass: { fileName: pressPass.fileName, isPrivate: false },
           bio
         };
 
         // Add optional fields if they exists
         favouriteArticle && (formData.favouriteArticle = favouriteArticle);
         jobTitle && (formData.jobTitle = jobTitle);
-        photoIDFile.fileName &&
```
Comment (RamyAlshurafa) [inline client/src/components/Pages/InternCreateProfile/index.js]: @ashatat can you explain this, I deleted, is it related to `pressPass` ? do I should return it back?

File: server/database/data/test/users.js [hunk 2]
```diff
@@ -22,36 +39,36 @@ module.exports = async () => {
...
+      account: orgAccount2._id,
     },
     {
       email: "brian@bbc.co.uk",
       name: "Brian Meyer",
       password: "123456",
       role: "organisation",
-      organisation: organisations[2],
+      organisation: organisations[2]._id,
       plan: "basic",
```
Comment (ashatat) [inline server/database/data/test/users.js]:
```suggestion
```
I don't think there is `plan` field anymore

File: server/database/models/Account.js [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+  donation: {
+    type: Number,
+    required: true,
+    default: 0,
+  },
+  // the total values of the current coupons that the org created and didnt used yet
+  couponsValue: {
+    type: Number,
+    default: 0,
+  },
```
Comment (ashatat) [inline server/database/models/Account.js]: what is the different between `donation`, `withdrawal` and `couponsValue` in terms of require: true or not?

File: server/database/models/Coupon.js [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+    type: Number,
+    required: true,
+    validate: {
+      validator: Number.isInteger,
+      message: "Used Days is not an integer value",
+    },
+  },
+  expirationDate: {
+    type: Date,
+    required: true,
```
Comment (ashatat) [inline server/database/models/Coupon.js]: I think we should add a validator here

File: server/database/models/Account.js (hunk 3 above)
Comment (RamyAlshurafa) [reply server/database/models/Account.js]: I think all of them require`required` field, I know some accounts these values will be `0`, but to make the query unified for all accounts

File: server/controllers/organisation/dashboard.js [hunk 5]
```diff
@@ -1,26 +1,18 @@
...
   try {
-    // get the organisation for that user
-    const [organisation] = await getUserOrg(_id);
-    if (!organisation) {
-      return next(boom.notFound());
-    }
-
-    // get dashboard data, [org details, notifications, interns]
-    const results = await dashboardQuery(organisation._id);
+    const results = await dashboardQuery(organisation);
```
[the remark this answers is not in the document]
Comment (RamyAlshurafa) [reply server/controllers/organisation/dashboard.js]: Good point, I added a check for the organisation field 
```js
  if (role !== "organisation" || !organisation) {
    return next(boom.unauthorized());
  }
```
For org's admin sign up we create an account and an organisation and make that user as an admin for that organisation, so this field always must be filled, but I added another check for more assurance

File: server/database/data/development/bookings.js [hunk 6]
```diff
@@ -9,48 +9,54 @@ module.exports = async () => {
...
+      host: listings[1].user,
       startDate: "2019-06-14",
       endDate: "2019-07-16",
       status: "confirmed",
-      payment: 200,
+      price: 200,
     },
     {
       listing: listings[2],
       user: interns[3],
```
Comment (ashatat) [inline server/database/data/development/bookings.js]:
```suggestion
      intern: interns[3],
```

**Comment by FarahZaqout** [comment]
@RamyAlshurafa 
hey, in the new bookings request, the output of the `createDatesArray` helper for the dates  `1-7-2019` to `4-7-2019` (values from the first test), is actually [`30-6-2019` to `3-7-2019`]. Is this intended, or is the helper bugged?

Also, the duplicate value test is testing values that are not duplicate, therefore it will not pass.

I updated the branch locally with test values that already exist in the database so it will pass. Let me know if you want me to push it.

**Comment by RamyAlshurafa** [comment]
Thanks @FarahZaqout 
the test was failing because the test was relying on hard-coded dates, so we changed all of them to dynamic dates
